validator/tests/integration/test_sawtooth_stats.py

```diff
@@ -1,69 +1,183 @@
     if os.environ.get("RUN_TEST_SUITES", False) == "1" else False
 
+class SawtoothStatsTestHarness():
+    '''
+    test_run()
+    - initializes stats config options
+    - creates instances of endpoint_manager, sawtooth stats,
+      sawtooth stats test module, sawtooth stats test harness,
+      and int key workload
+    - initializes test_run_in_thread() in reactor thread
+    - starts reactor
+    test_run_in_thread()
+    - sequences the stats test operations
+    - collects stats on initial network state; verifies they are correct
+    - runs int key workload to increment txn and block count
+    - collects stats on new network state; verifies they are correct
+    - note: sleeps after calls to endpoint_discovery_loop() and stats_loop()
+      to allow time for associated http requests to validators can complete
+    - Note: runs in reactor thread so that blocking calls to time.sleep()
+      do not suspend main reactor thread required to complete http calls
+    '''
+    def __init__(self):
+        self.epm = None
+        self.ss = None
+        self.sst = None
+        self.int_key_load_test = None
+        self.urls = None
+
+        self.keys = 10
+        self.rounds = 2
+
+    def test_run(self, urls, config_opts=None, config_dict=None):
 
-@unittest.skipUnless(RUN_TEST_SUITES, "Must be run in a test suites")
-class TestSawtoothStats(unittest.TestCase):
-    def __init__(self, test_name, urls=None):
-        super(TestSawtoothStats, self).__init__(test_name)
         self.urls = urls
+        self.url = urls[0]
 
-    def test_sawtooth_stats(self):
-        try:
+        config = self.default_config()
+
+        # update defaults with cli options if provided
+        if config_opts is None:
+            config['EndpointManager']['urls'] = self.urls
+        else:
+            self.update_config(config, config_opts)
+
+        if config_dict is not None:
+            self.update_config_dict(config_dict, config)
+
+        self.epm = EndpointManager(config['EndpointManager']['urls'])
+        self.ss = SawtoothStats(self.epm, config)
+        self.sst = SawtoothStatsTest(None, None)
+        self.sst.initialize(self.ss.stats_modules)
+
+        reactor.callInThread(self.test_run_in_thread)
+
+        print "test complete with no errors"
+
+    def _are_registered(self, timeout=256, validators=5):
+        registered = False
+        wait_time = 0
+        while not registered:
+            reactor.callFromThread(self.epm.endpoint_discovery_loop)
+            time.sleep(1)
+            if len(self.epm.endpoint_urls) is validators:
+                registered = True
+            if wait_time > timeout:
+                break
+            wait_time += 1
+        return registered
+
+    def test_run_in_thread(self):
+        # get list of validators in network
+        # sleep to wait for requests to complete
+
+        assert self._are_registered(), "max registration time exceeded"
+
+        assert len(self.epm.endpoint_urls) == 5
```

Comment:
this is to make sure we have the right number of validators that were lunched with Sawtooth_suites